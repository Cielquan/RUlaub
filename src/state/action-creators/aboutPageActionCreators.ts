import { Dispatch } from "redux";

import { AboutPageActionType } from "../action-types";
import { AboutPageAction } from "../actions";

export const openAboutPageAction = (): AboutPageAction => ({
  type: AboutPageActionType.OPEN,
});

export const openAboutPage =
  () =>
  (dispatch: Dispatch<AboutPageAction>): void => {
    dispatch(openAboutPageAction());
  };

export const closeAboutPageAction = (): AboutPageAction => ({
  type: AboutPageActionType.CLOSE,
});

export const closeAboutPage =
  () =>
  (dispatch: Dispatch<AboutPageAction>): void => {
    dispatch(closeAboutPageAction());
  };
