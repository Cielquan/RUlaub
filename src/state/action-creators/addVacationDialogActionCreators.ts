import { Dispatch } from "redux";

import { AddVacationDialogActionType } from "../action-types";
import { AddVacationDialogAction } from "../actions";

export const openAddVacationDialogAction = (): AddVacationDialogAction => ({
  type: AddVacationDialogActionType.OPEN,
});

export const openAddVacationDialog =
  () =>
  (dispatch: Dispatch<AddVacationDialogAction>): void => {
    dispatch(openAddVacationDialogAction());
  };

export const closeAddVacationDialogAction = (): AddVacationDialogAction => ({
  type: AddVacationDialogActionType.CLOSE,
});

export const closeAddVacationDialog =
  () =>
  (dispatch: Dispatch<AddVacationDialogAction>): void => {
    dispatch(closeAddVacationDialogAction());
  };
