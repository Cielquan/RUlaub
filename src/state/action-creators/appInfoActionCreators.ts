import { Dispatch } from "redux";
import { AppInfoType } from "../action-types";
import { AppInfoAction } from "../actions";

export const changeAppInfoName =
  (newName: string) =>
  (dispatch: Dispatch<AppInfoAction>): void => {
    dispatch({
      type: AppInfoType.NAME,
      payload: newName,
    });
  };

export const changeAppInfoVersion =
  (newVersion: string) =>
  (dispatch: Dispatch<AppInfoAction>): void => {
    dispatch({
      type: AppInfoType.VERSION,
      payload: newVersion,
    });
  };

export const changeAppInfoInfoText =
  (newInfoText: string) =>
  (dispatch: Dispatch<AppInfoAction>): void => {
    dispatch({
      type: AppInfoType.INFO_TEXT,
      payload: newInfoText,
    });
  };
