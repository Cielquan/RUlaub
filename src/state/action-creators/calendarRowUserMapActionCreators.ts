import { Dispatch } from "redux";
import { UsersDataSchema } from "../../types/usersData.schema";

import { CalendarRowUserMapActionType } from "../action-types";
import { CalendarRowUserMapAction } from "../actions";

export const updateCalendarRowUserMapAction = (
  payload: UsersDataSchema
): CalendarRowUserMapAction => ({
  type: CalendarRowUserMapActionType.UPDATE,
  payload,
});

export const updateCalendarRowUserMap =
  (payload: UsersDataSchema) =>
  (dispatch: Dispatch<CalendarRowUserMapAction>): void => {
    dispatch(updateCalendarRowUserMapAction(payload));
  };
