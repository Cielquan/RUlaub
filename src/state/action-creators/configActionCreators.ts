import { invoke } from "@tauri-apps/api";
import { open, save } from "@tauri-apps/api/dialog";
import { ProviderContext } from "notistack";
import { dirname } from "path";
import { Dispatch } from "redux";

import getErrorCatalogueMsg from "../../backendAPI/errorMsgCatalogue";
import {
  ConfigFileSchema as ConfigFile,
  LogLevel,
  SupportedLanguages,
  SupportedThemes,
} from "../../backendAPI/types/configFile.schema";
import { validateConfig } from "../../backendAPI/validation";
import { enqueuePersistendErrSnackbar } from "../../utils/snackbarUtils";
import { ConfigActionType } from "../action-types";
import { ConfigAction } from "../actions";

export const updateConfigAction = (payload: ConfigFile): ConfigAction => ({
  type: ConfigActionType.UPDATE,
  payload,
});

export const loadConfig =
  () =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    const data = await invoke("get_config_state");

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-loadConfig",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setLanguage =
  (lang: SupportedLanguages, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_langauge", { lang });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setLanguage",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setLogLevel =
  (level: LogLevel, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_log_level", { level });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setLogLevel",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setTheme =
  (theme: SupportedThemes, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_theme", { theme });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setTheme",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setTodayAutoscrollLeftOffset =
  (offset: number, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_today_autoscroll_left_offset", {
        offset,
      });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setTodayAutoscrollLeftOffset",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setUserName =
  (name: string, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_user_name", { name });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setUserName",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setYearChangeScrollBegin =
  (doScroll: boolean, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_year_change_scroll_begin", { doScroll });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setYearChangeScrollBegin",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const setYearToShow =
  (year: number, snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    let data;
    try {
      data = await invoke<ConfigFile>("set_year_to_show", { year });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-setYearToShow",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

const FILTERS = [{ name: "Database", extensions: ["db"] }];

export const createNewDB =
  (snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    const path = await save({ filters: FILTERS });
    if (path === null) return;

    let data;
    try {
      data = await invoke<ConfigFile>("create_db", { path });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-createNewDB",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };

export const selectDB =
  (snackbarHandles: ProviderContext) =>
  async (dispatch: Dispatch<ConfigAction>): Promise<void> => {
    const dbUri = (await invoke<ConfigFile>("get_config_state")).settings.databaseUri;

    const path = (await open({
      defaultPath: dbUri ? dirname(dbUri) : undefined,
      filters: FILTERS,
      multiple: false,
      directory: false,
    })) as string;
    if (path === null) return;

    let data;
    try {
      data = await invoke<ConfigFile>("set_db_uri", { path });
    } catch (err) {
      enqueuePersistendErrSnackbar(getErrorCatalogueMsg(err as string), snackbarHandles);
      return;
    }

    let conf;
    try {
      conf = await validateConfig(data);
    } catch (err) {
      invoke("log_error", {
        target: "config",
        message: `Config data validation failed: ${err}`,
        location: "state/action-creators/configActionCreators.ts-selectDB",
      });
      return;
    }
    dispatch(updateConfigAction(conf));
  };
