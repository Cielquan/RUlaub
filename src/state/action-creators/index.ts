export * from "./configActionCreators";
export * from "./infoPageActionCreators";
export * from "./publicHolidaysDataActionCreators";
export * from "./publicHolidaysDialogActionCreators";
export * from "./schoolHolidaysDataActionCreators";
export * from "./settingsDialogActionCreators";
export * from "./sideMenuActionCreators";
export * from "./vacationDataActionCreators";
