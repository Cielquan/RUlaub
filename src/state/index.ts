export * as actionCreators from "./action-creators";
export { default as store } from "./store";
export * from "./reducers";
