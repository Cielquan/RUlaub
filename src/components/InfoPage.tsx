import { Trans } from "@lingui/macro";
import { Info as InfoIcon } from "@mui/icons-material";
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Link,
  Slide,
  Typography,
} from "@mui/material";
import { TransitionProps } from "@mui/material/transitions";
import { Box } from "@mui/system";
import React, { forwardRef, ReactElement } from "react";
import { useDispatch, useSelector } from "react-redux";
import { bindActionCreators } from "redux";

import { actionCreators, State } from "../state";

import * as pjson from "../../package.json";

const Transition = forwardRef(
  (
    props: TransitionProps & { children?: ReactElement },
    ref: React.Ref<unknown>
    // eslint-disable-next-line react/jsx-props-no-spreading
  ) => <Slide direction="up" ref={ref} {...props} />
);
Transition.defaultProps = {
  children: <></>,
};

interface Props {
  onClick?: () => void;
}

const InfoPage = ({ onClick }: Props): ReactElement => {
  const dispatch = useDispatch();
  const { closeInfoPage } = bindActionCreators(actionCreators, dispatch);
  const infoPageState = useSelector((state: State) => state.infoPage);

  const ghLink = (
    <Link href="https://github.com/Cielquan/RUlaub" target="_blank" rel="noreferrer">
      github
    </Link>
  );

  const id = "info-page";

  return (
    <Dialog
      aria-labelledby={id}
      data-testid={id}
      open={infoPageState}
      TransitionComponent={Transition}
      keepMounted
      onClose={closeInfoPage}
    >
      <DialogTitle id={id} sx={{ display: "flex", alignItems: "center" }}>
        <Box sx={{ flexGrow: 1 }}>{`RUlaub v${pjson.version}`}</Box>
        <InfoIcon />
      </DialogTitle>
      <DialogContent>
        <DialogContentText sx={{ whiteSpace: "pre-wrap" }}>
          <Typography gutterBottom component="span">
            <Trans>
              RUlaub is licensed under either &apos;Apache License 2.0&apos; or
              &apos;MIT License&apos; at your option.
            </Trans>
          </Typography>
          <Typography component="span">
            <Trans>
              For the Source Code, Documentation and copies of the License files please
              see the github repository at: {ghLink}.
            </Trans>
          </Typography>
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          data-testid={`${id}-btn`}
          onClick={() => {
            if (typeof onClick === "function") onClick();
            closeInfoPage();
          }}
          autoFocus
        >
          <Trans>Close</Trans>
        </Button>
      </DialogActions>
    </Dialog>
  );
};
InfoPage.defaultProps = {
  onClick: () => undefined,
};

export default InfoPage;
