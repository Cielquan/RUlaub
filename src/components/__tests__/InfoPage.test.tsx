import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { Provider } from "react-redux";
import { CombinedState } from "redux";
import { MockStoreEnhanced } from "redux-mock-store";

import Languages from "../../i18n";
import InfoPage from "../InfoPage";
import { initialState, State } from "../../state";
import generateMockStore from "../../testUtils";
import { closeInfoPageAction } from "../../state/action-creators";

describe("InfoPage component", () => {
  let testState: State;
  let mockStore: MockStoreEnhanced<Partial<CombinedState<State>>>;

  beforeEach(() => {
    testState = JSON.parse(JSON.stringify(initialState));
    mockStore = generateMockStore(testState);
  });

  it("renders invisible by default", () => {
    expect(mockStore.getState().language).toEqual(Languages.english);
    expect(mockStore.getState().infoPage).toBe(false);
    render(
      <Provider store={mockStore}>
        <I18nProvider i18n={i18n}>
          <InfoPage />
        </I18nProvider>
      </Provider>
    );
    expect(screen.getByTestId("info-page")).toMatchSnapshot();
    expect(screen.getByTestId("info-page")).toHaveAttribute("aria-hidden", "true");
    expect(screen.getByTestId("info-page")).toHaveAttribute(
      "style",
      expect.stringMatching(/.*visibility: hidden.*/i)
    );
  });

  it("renders correctly when open", () => {
    expect(mockStore.getState().language).toEqual(Languages.english);
    testState.infoPage = true;
    const alteredMockStore = generateMockStore(testState);
    expect(alteredMockStore.getState().infoPage).toBe(true);
    render(
      <Provider store={mockStore}>
        <I18nProvider i18n={i18n}>
          <InfoPage />
        </I18nProvider>
      </Provider>
    );
    expect(screen.getByTestId("info-page")).toMatchSnapshot();
    expect(screen.getByTestId("info-page")).not.toHaveAttribute("aria-hidden");
    expect(screen.getByTestId("info-page")).not.toHaveAttribute(
      "style",
      expect.stringMatching(/.*visibility: hidden.*/i)
    );
  });

  it("closes when button is clicked", () => {
    testState.infoPage = true;
    const alteredMockStore = generateMockStore(testState);
    const onClickMock = jest.fn();

    expect(alteredMockStore.getState().infoPage).toBe(true);
    render(
      <Provider store={alteredMockStore}>
        <I18nProvider i18n={i18n}>
          <InfoPage onClick={onClickMock} />
        </I18nProvider>
      </Provider>
    );
    expect(screen.getByTestId("info-page")).not.toHaveAttribute("aria-hidden");
    expect(screen.getByTestId("info-page")).not.toHaveAttribute(
      "style",
      expect.stringMatching(/.*visibility: hidden.*/i)
    );

    userEvent.click(screen.getByTestId("info-page-btn"));
    expect(onClickMock).toHaveBeenCalledTimes(1);
    const expectedActions = [closeInfoPageAction()];
    expect(alteredMockStore.getActions()).toEqual(expectedActions);
  });
});
