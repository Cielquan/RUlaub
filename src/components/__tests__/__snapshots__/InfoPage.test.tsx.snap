// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> renders correctly when open 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="info-page"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="info-page"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-tqj0yt-MuiTypography-root-MuiDialogTitle-root"
        id="info-page"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          RUlaub v1.0.0
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-e901ri-MuiTypography-root-MuiDialogContentText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            RUlaub is licensed under either 'Apache License 2.0' or 'MIT License' at your option.
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            For the Source Code, Documentation and copies of the License files please see the github repository at: 
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="https://github.com/Cielquan/RUlaub"
              rel="noreferrer"
              target="_blank"
            >
              github
            </a>
            .
          </span>
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-focusVisible css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="info-page-btn"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          >
            <span
              class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`<AboutPage /> renders invisible by default 1`] = `
<div
  aria-hidden="true"
  class="MuiModal-root MuiModal-hidden MuiDialog-root css-ew73zq-MuiModal-root-MuiDialog-root"
  data-testid="info-page"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 0; visibility: hidden;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="visibility: hidden; webkit-transform: translateY(768px); transform: translateY(768px);"
    tabindex="-1"
  >
    <div
      aria-labelledby="info-page"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-tqj0yt-MuiTypography-root-MuiDialogTitle-root"
        id="info-page"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          RUlaub v1.0.0
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiDialogContentText-root MuiTypography-root MuiTypography-body1 css-e901ri-MuiTypography-root-MuiDialogContentText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            RUlaub is licensed under either 'Apache License 2.0' or 'MIT License' at your option.
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            For the Source Code, Documentation and copies of the License files please see the github repository at: 
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="https://github.com/Cielquan/RUlaub"
              rel="noreferrer"
              target="_blank"
            >
              github
            </a>
            .
          </span>
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-focusVisible css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="info-page-btn"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          >
            <span
              class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
